Formula/a/asdf.rb

```diff
@@ -16,38 +17,40 @@ class Asdf < Formula
     sha256 cellar: :any_skip_relocation, all: "a94dd362ce5c3a818f4fa56607eac3c616a3c1191e9f41480093bda2dc308af4"
   end
 
-  depends_on "autoconf"
-  depends_on "automake"
-  depends_on "coreutils"
-  depends_on "libtool"
-  depends_on "libyaml"
+  depends_on "go" => :build
+
   depends_on "openssl@3"
-  depends_on "readline"
-  depends_on "unixodbc"
 
   def install
-    bash_completion.install "completions/asdf.bash" => "asdf"
-    fish_completion.install "completions/asdf.fish"
-    zsh_completion.install "completions/_asdf"
-    libexec.install Dir["*"]
+    system("go", "build", *std_go_args(ldflags: "-s -X main.version=#{version}"), "./cmd/asdf")
+    original_stdout=$stdout
+
+    $stdout = File.new("asdf.bash", "w")
+    system(bin/"asdf", "completion", "bash")
+    bash_completion.install "asdf.bash" => "asdf"
```

Comment:
`asdf completion bash` prints the completion code to STDOUT, and I want to redirect it to a file so that I can install it with `bash_completion.install`. Setting `$stdout` works, but should I be using the `shell_output` method instead?

Reply:
You're probably looking for `generate_completions_from_executable`.

Reply:
Thanks! That's much nicer! PR updated.